api/src/main/java/org/apache/iceberg/actions/MigrateTable.java

```diff
@@ -41,6 +41,14 @@ public interface MigrateTable extends Action<MigrateTable, MigrateTable.Result>
    */
   MigrateTable tableProperty(String name, String value);
 
+  /**
+   * Drops the backup of the original table after a successful migration
+   *
+   * @param dropBackup if true, removes the backup table
+   * @return this for method chaining
+   */
+  MigrateTable dropBackup(boolean dropBackup);
```

Comment:
@danielcweeks @rdblue wanted to check as its api module, we can add a new method if all subclass implement it, right?  Debating whether or not to make a default.

Reply:
I checked with @danielcweeks offline, we could put a default that throws UnsupportedOperation or just leave it as is.  Probably the risk is small anyone implements it, so i think we are good.